README.md

```diff
@@ -105,6 +105,28 @@ make manifests
 
 More information can be found via the [Kubebuilder Documentation](https://book.kubebuilder.io/introduction.html)
 
+## Logging
+Logs are following the general guidelines: 
+- `logger.Info()` describe a high-level state of the resource such as creation, deletion and which reconciliation path was taken.  
+- `logger.Error()` describe only those errors that are not returned in the result of the reconciliation. If error is occurred there should be only one error message. 
+- `logger.V(1).Info()` debug level logs to give information about every change or event caused by the resource as well as every update of the resource.
+
+The `--zap-devel` argument will enable debug level logs for the output. Otherwise, all `V()` logs are ignored.  
+To query logs locally you can use `jq`. For example:
+Retrieve logs by 
+```shell
+kubectl get deployments -l app.kubernetes.io/part-of=dns-operator -A
+
+NAMESPACE             NAME                              READY 
+dns-operator-system   dns-operator-controller-manager   1/1   
+```
+And query them. For example:
+```shell
+kubectl logs -l control-plane=dns-operator-controller-manager -n dns-operator-system --tail -1 | sed '/^{/!d' | jq 'select(.controller=="dnsrecord" and .level=="info")'
+```
+You could use selector in the `jq` with `and`/`not`/`or` to restrict. 
+
```

Comment:
Should we not list here the common metadata that can be attached to a log statement and what they mean (ownerID, zoneID etc..), anything we add at different parts of the reconcile using [log.WithValues](https://github.com/Kuadrant/dns-operator/blob/main/internal/controller/dnsrecord_controller.go#L124-L125) ? Maybe a generic example of how you can filter logs using jq for any of these or a combination of values.  I kind of thought that was the main point of #117 (Maybe I'm wrong).

I'm less concerned about the testing it in an aggregator part of that issue, if the logs are structured (which they are), and the metadata is on the requests then they can be filtered using whatever tool you choose to use for log aggregation and querying.

I'm not sure this PR is doing enough to cover the first two asks of #117 though:

* Log metadata defined and documented
* Logging reviewed in the controllers and providers updated

Reply:
Added section with `common metadata` with some of the keys (not all of them as maintaining it doesn't sound feasible) and another example of `jq` usage.

Reply:
The task also says "logs metadata defined and documented". The documentation seems straightforward, but with "defined" - all the important ones (the ones that will allow to narrow down logs) are there anyway. Reconcile ID and CR name will be added anyway so I'm not sure how else we could "define" them :thinking: